src/data-workspace/indicators-table-body/use-indicator-value.js

```diff
@@ -43,33 +42,35 @@ export const useIndicatorValue = ({
         [denominator, numerator]
     )
 
+    const affectedValues = [...affectedDataElementsLookup]
+        .map((de) =>
+            Object.values(dataValues[de] ?? {})
+                .sort((a, b) =>
+                    a?.categoryOptionCombo?.localeCompare(
+                        b?.categoryOptionCombo
+                    )
+                )
+                .map((d) => d.value)
+        )
+        .flat()
+    const stringifiedValues = JSON.stringify(affectedValues)
+
     return useMemo(() => {
         /*
          * Only recompute the indicator value when a field related to
          * a data element in the indicator expression template is blurred.
          */
-        const { dataElementId } = parseFieldOperand(blurredField)
-        const containsAffectedDataElement =
-            affectedDataElementsLookup.has(dataElementId)
 
-        if (indicatorValueRef.current === null || containsAffectedDataElement) {
-            indicatorValueRef.current = computeIndicatorValue({
-                denominator,
-                numerator,
-                factor,
-                dataValues,
-                decimals,
-            })
-        }
+        indicatorValueRef.current = computeIndicatorValue({
+            denominator,
+            numerator,
+            factor,
+            dataValues,
+            decimals,
+        })
 
         return indicatorValueRef.current
-    }, [
-        affectedDataElementsLookup,
-        blurredField,
-        denominator,
-        factor,
-        dataValues,
-        numerator,
-        decimals,
-    ])
+    }, [stringifiedValues, denominator, numerator, factor, decimals])
+    // dependency array is non-exhaustive insofar as dataValues is excluded
```

Comment:
I went with getting a simple representation of the values and putting this in the dependency array (which makes the dependency array non-exhaustive). An alternative would be to stringify/parse a longer string representation of dataValues, but I feel like that's less efficient, so I don't think it's worth it to get rid of dependency array warning. I also don't want to turn off the eslint warning for the non-exhaustive dependencies, so we don't edit it and potentially get even less exhaustive in the future.